### ISSUE TITLE
Missing DeviceCodeAuthorizationCodeExpired error in MsalClientException

### ISSUE THREAD

**Issue by jmprieur** [AzureAD/microsoft-authentication-library-for-dotnet#638]
**Which Version of MSAL are you using ?**
MSAL.NET 2.2.0-preview

**Which platform has the issue?**
all supporting Device Code flow.

**What authentication flow has the issue?**
* Desktop 
    * [ ] Interactive
    * [ ] WIA
    * [ ] U/P
    * [ x] Device code flow (browserless)
* [ ] Mobile
* Web App
    * [ ] Authorization code
    * [ ] OBO
* Web API
    * [ ] OBO

**Repro**
Write code using Device Code flow
try to catch the case where the user has not signed-in during the first 15 mins
MSAL will throw an `MsalClientException`, and the string containing the error code is "Verification code expired before contacting the server".

```csharp
            AuthenticationResult result;
            try
            {
                result = await App.AcquireTokenWithDeviceCodeAsync(scopes,
                    deviceCodeCallback =>
                    {
                        Console.WriteLine(deviceCodeCallback.Message);
                        return Task.FromResult(0);
                    });
            }

            catch (MsalClientException ex)
            {
                string errorCode = ex.ErrorCode;

                // "Verification code expired before contacting the server"
                // This exception will occur if the user does not manage to sign-in before a time out (15 mins) and the
                // call to `AcquireTokenWithDeviceCodeAsync` is not cancelled in between
                result = null;
            }
            return result;
        }
```

**Expected behavior**
`MsalClientException`  has a constant named `DeviceCodeAuthorizationCodeExpired` to process this error code so that I compare it in the catch block for `MsalClientException`

**Actual behavior**
No constant is defined in `MsalClientException` for this case in the platforms where DeviceCodeFlow is supported. Note that ADAL.NET defines 

**Possible Solution**
Add a static string named `DeviceCodeAuthorizationCodeExpired` to `MsalClientException`

**Comment by jmprieur** [comment]
cc: @bgavrilMS

**Comment by bgavrilMS** [comment]
There seem to be bunch of error codes that we do not expose in MSAL. Should be a simple fix.

**Comment by jennyf19** [comment]
fixed w/[2.30](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/releases/tag/MSAL-v2.3.0)
